examples/platform/nxp/se05x/DeviceAttestationSe05xCredsExample_v2.cpp

```diff
@@ -172,49 +173,50 @@ CHIP_ERROR ExampleSe05xDACProviderv2::SignWithDeviceAttestationKey(const ByteSpa
 
-    msg_reader.Init(tlv_buffer);
-    msg_reader.Next();
+    msg_reader.Init(message_to_sign);
 
     /* To be removed. Use common key id to sign message */
     static bool sign_cert_decl_attest = 1;
 
     if (sign_cert_decl_attest)
     {
-        /* Skip certificate declaration tag */
-        msg_reader.Next();
-
-        tagValue = msg_reader.GetTag();
-        VerifyOrReturnError(TLV::ContextTag(2) == tagValue, CHIP_ERROR_INVALID_TLV_TAG);
-        /* Get attestation nonce */
-        ByteSpan attest_nonce;
-        msg_reader.Get(attest_nonce);
-        /* Set attestation nonce */
-        VerifyOrReturnError(CHIP_NO_ERROR ==
-                                se05xSetCertificate(CD_ATTEST_NONCE_DATA_SE05X_ID, attest_nonce.data(), attest_nonce.size()),
-                            CHIP_ERROR_INTERNAL);
-
-        msg_reader.Next();
-        tagValue = msg_reader.GetTag();
-        VerifyOrReturnError(TLV::ContextTag(3) == tagValue, CHIP_ERROR_INVALID_TLV_TAG);
-        ByteSpan time_stamp;
-        msg_reader.Get(time_stamp);
-        uint8_t tmp = time_stamp.size();
+        /* Get length and Skip certificate declaration tag */
+        ReturnErrorOnFailure(TLV::Utilities::Find(msg_reader, TLV::ContextTag(1), tagReader));
+        uint8_t cdlen = tagReader.GetLength();
+
+        ReturnErrorOnFailure(TLV::Utilities::Find(msg_reader, TLV::ContextTag(2), tagReader));
+        uint8_t attlen = tagReader.GetLength();
+        if (attlen > 0)
+        {
+            /* Get attestation nonce */
+            ByteSpan attest_nonce;
+            ReturnErrorOnFailure(tagReader.Get(attest_nonce));
+            /* Set attestation nonce */
+            VerifyOrReturnError(CHIP_NO_ERROR ==
+                                    se05xSetCertificate(CD_ATTEST_NONCE_DATA_SE05X_ID, attest_nonce.data(), attest_nonce.size()),
+                                CHIP_ERROR_INTERNAL);
+        }
+
+        ReturnErrorOnFailure(TLV::Utilities::Find(msg_reader, TLV::ContextTag(3), tagReader));
+        uint8_t tslen = tagReader.GetLength();
         /* Set time stamp length */
-        VerifyOrReturnError(CHIP_NO_ERROR == se05xSetCertificate(CD_TIME_STAMP_LEN_SE05X_ID, &tmp, 1), CHIP_ERROR_INTERNAL);
-        if (time_stamp.size() > 0)
+        VerifyOrReturnError(CHIP_NO_ERROR == se05xSetCertificate(CD_TIME_STAMP_LEN_SE05X_ID, &tslen, 1), CHIP_ERROR_INTERNAL);
+        if (tslen > 0)
         {
+            ByteSpan time_stamp;
+            tagReader.Get(time_stamp);
             /* Set time stamp data */
             VerifyOrReturnError(CHIP_NO_ERROR ==
                                     se05xSetCertificate(CD_TIME_STAMP_DATA_SE05X_ID, time_stamp.data(), time_stamp.size()),
                                 CHIP_ERROR_INTERNAL);
         }
 
-        if (message_to_sign.size() >= 16)
+        if (message_to_sign.size() -
+                (cdlen + attlen + tslen + 9 /* Tag + control byte + len */ + 2 /* start and end containers*/) >=
```

Comment:
Why wouldn't you just use the tag reader's actual position (after calling `Next()` on it) here instead of trying to recompute it?

Reply:
I see some issues when i use the Next() for parsing the TLV data here. Using Find() utility to get the TLV data. For this I am  recomputing the value data utilized.

Reply:
Updated the code to use GetRemainingLength() now.